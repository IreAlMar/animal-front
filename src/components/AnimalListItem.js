import React from 'react';
import PropTypes from 'prop-types';

const AnimalListItem = ({
  animal,
  favorited,
  onClick,
  onFavorited,
}) => (

    <li
      className="py2 border-bottom border-bottom-dashed pointer"
      // key={animal.taxonid}
      onClick={() => onClick(animal.scientific_name)}
    >
      <span
        className="mr1"
        onClick={e => {
          // To rpevent the animal to be shown when box clicked
          e.stopPropagation();
          onFavorited(animal.taxonid)
        }}
        role="img"
        aria-label="favorite"
      >
        {favorited ? '✅' : '☐'}
      </span>
      <span>{animal.scientific_name}</span>
      <span>{animal.category}</span>
    </li>
  );

AnimalListItem.propTypes = {
  animal: PropTypes.object,
  favorited: PropTypes.bool,
  onClick: PropTypes.func,
  onFavorited: PropTypes.func,
};

export default AnimalListItem;