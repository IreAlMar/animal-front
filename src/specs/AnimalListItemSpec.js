import React from 'react';
import renderer from 'react-test-renderer';
import AnimalListItem from '../components/AnimalListItem';

const testAnimal = {
  scientific_name: 'Chamaleon test name',
  taxonid: 1,
  category: 'Test category',
};

describe('<AnimalListItem />', () => {
  // AnimalListItem takes the following props: ({ animal, favorited, onClick, onFavorited })
  // So test what happens in each case:
  test('Should not break if no animal is passed', () => {
    const component = renderer.create(<AnimalListItem />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('Should correctly reder animal', () => {
    const component = renderer.create(<AnimalListItem animal={testAnimal} />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('Should render favorite state', () => {
    // favorited={true} -> Value must be omitted for boolean attribute seslint
    const component = renderer.create(<AnimalListItem animal={testAnimal} favorited />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
