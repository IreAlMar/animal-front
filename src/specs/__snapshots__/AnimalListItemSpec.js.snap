// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AnimalListItem /> Should correctly reder animal 1`] = `
<li
  className="py2 border-bottom border-bottom-dashed pointer"
  onClick={[Function]}
>
  <span
    aria-label="favorite"
    className="mr1"
    onClick={[Function]}
    role="img"
  >
    ☐
  </span>
  <span>
    Chamaleon test name
  </span>
  <span>
    Test category
  </span>
</li>
`;

exports[`<AnimalListItem /> Should not break if no animal is passed 1`] = `
<li
  className="py2 border-bottom border-bottom-dashed pointer"
  onClick={[Function]}
>
  <span
    aria-label="favorite"
    className="mr1"
    onClick={[Function]}
    role="img"
  >
    ☐
  </span>
  <span />
  <span />
</li>
`;

exports[`<AnimalListItem /> Should render favorite state 1`] = `
<li
  className="py2 border-bottom border-bottom-dashed pointer"
  onClick={[Function]}
>
  <span
    aria-label="favorite"
    className="mr1"
    onClick={[Function]}
    role="img"
  >
    ✅
  </span>
  <span>
    Chamaleon test name
  </span>
  <span>
    Test category
  </span>
</li>
`;
